CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+# How to contribute
+
+This document is liberally copied/based on the same in
+[Puppet](https://github.com/puppetlabs/puppet).
+
+Third-party patches, as well as patches from those within Puppet who don't
+maintain this project directly, are essential for keeping Puppet Agent
+great. We simply can't access the huge number of platforms and myriad
+configurations for running Puppet Agent. We want to keep it as easy as possible
+to contribute changes that get things working in your environment. There are a
+few guidelines that we need contributors to follow so that we can have a chance
+of keeping on top of things.
+
+
+## Getting Started
+
+* Make sure you have a [Jira account](https://tickets.puppetlabs.com)
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue, assuming one does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub
+
+## Making Changes
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * Only target release branches if you are certain your fix must be on that
+    branch.
+  * To quickly create a topic branch based on master; `git checkout -b
+    fix/master/my_contribution master`. Please avoid working directly on the
+    `master` branch.
+* Make commits of logical units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the proper format.
+
+````
+    (PA-1234) Make the example in CONTRIBUTING imperative and concrete
+
+    Without this patch applied the example commit message in the CONTRIBUTING
+    document is not a concrete example.  This is a problem because the
+    contributor is left to imagine what the commit message should look like
+    based on a description rather than an example.  This patch fixes the
+    problem by making the example concrete and imperative.
+
+    The first line is a real life imperative statement with a ticket number
+    from our issue tracker.  The body describes the behavior without the patch,
+    why this is a problem, and how the patch fixes the problem when applied.
+````
+
+* Make sure you have added the necessary tests for your changes.
+* Run _all_ the tests to assure nothing else was accidentally broken.
```

Comment:
Oh that's interesting. Do we really not even have specs for puppet-agent? I assumed there was some part of https://jenkins.puppetlabs.com/view/puppet-agent/view/manual/view/stable/ that actually did some testing, but it appears not..